src/test/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSourceTest.java

```diff
@@ -146,6 +159,25 @@ public void prepareMockGitHub() throws Exception {
         source = new GitHubSCMSource(null, "http://localhost:" + githubApi.port(), GitHubSCMSource.DescriptorImpl.SAME, null, "cloudbeers", "yolo");
     }
 
+    @Test
+    @Issue("JENKINS-48035")
+    public void testGitHubRepositoryNameContributor() throws IOException {
```

Comment:
and test negative case (e.g. a `GitScmSource` (and make it point to not-github) not `GitHubSCMSource` ) ... and test when there is a regular job like freestyle that doesn't own an SCMSource